crates/sui-gql-client/src/queries/filtered_full_objects.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+
+use af_sui_types::{Address, Object, ObjectId};
+use futures::Stream;
+
+use super::fragments::{ObjectFilterV2, PageInfoForward};
+use crate::queries::Error;
+use crate::{extract, missing_data, scalars, schema, GraphQlClient, GraphQlResponseExt, Paged};
+
+pub(super) fn query<C: GraphQlClient>(
+    client: &C,
+    owner: Option<Address>,
+    type_: Option<String>,
+    page_size: Option<u32>,
+) -> impl Stream<Item = Result<(ObjectId, Object), Error<C::Error>>> + '_ {
+    async_stream::try_stream! {
+        let filter = ObjectFilterV2 {
+            owner,
+            type_,
+            ..Default::default()
+        };
+        let mut vars = Variables {
+            after: None,
+            first: page_size.map(|v| v.try_into().unwrap_or(i32::MAX)),
+            filter: Some(filter),
+        };
+        let mut has_next_page = true;
+        while has_next_page {
+            let (page_info, objects) = request(client, vars.clone()).await?;
+
+            vars.after = page_info.end_cursor.clone();
+            has_next_page = page_info.has_next_page;
+
+            for value in objects {
+                yield value;
+            }
+        }
+    }
+}
+
+async fn request<C: GraphQlClient>(
+    client: &C,
+    vars: Variables,
+) -> Result<
+    (
+        PageInfoForward,
+        impl Iterator<Item = (ObjectId, Object)> + 'static,
+    ),
+    Error<C::Error>,
+> {
+    let response = client
+        .query::<Query, _>(vars)
+        .await
+        .map_err(Error::Client)?;
+    let data = response.try_into_data()?;
+
+    let ObjectConnection { nodes, page_info } = extract!(data?.objects);
+    let mut raw_objs = HashMap::new();
+    for ObjectGql { id, object } in nodes.into_iter() {
+        let wrapped = object.ok_or(missing_data!("Bcs for object {id}"))?;
+        raw_objs.insert(id, wrapped.into_inner());
+    }
+    Ok((page_info, raw_objs.into_iter()))
+}
```

Comment:
```suggestion
async fn request<C: GraphQlClient>(
    client: &C,
    vars: Variables,
) -> Result<
    (
        PageInfoForward,
        impl Iterator<Item = Result<Object, Error<C::Error>>> + 'static,
    ),
    Error<C::Error>,
> {
    let response = client
        .query::<Query, _>(vars)
        .await
        .map_err(Error::Client)?;
    let data = response.try_into_data()?;

    let ObjectConnection { nodes, page_info } = extract!(data?.objects);
    let raw_objs = nodes.into_iter()
        .map(|ObjectGql { id, object }| {
            object.ok_or(missing_data!("Bcs for object {id}"))
        })
    Ok((page_info, raw_objs))
}
```
Then above in the `query`
```rust
            for object in objects {
                yield object?;
            }
```